### PR TITLE
Publish unitests results to the pull requests

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/1234

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -150,12 +150,11 @@ jobs:
...
           sudo make gotest | tee test-results/junit/unittests.out
-      - name: Upload Unit Test Results
+      - name: Publish Unit Test Results
         if: always()
-        uses: actions/upload-artifact@v1
+        uses: EnricoMi/publish-unit-test-result-action@v1.9
         with:
-          name: unittests-results
-          path: test-results/junit/unittests.xml
+          files: test-results/junit/unittests.xml
```
Comment (tigrannajaryan) [inline .github/workflows/build-and-test.yml:157]: What does this do? Is it supposed to publish the results somewhere visible?

Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:157]: Trying to figure that out, for the moment there is a permission issue that I hope I fixed

**Review by gramidt — APPROVED** [review]
The Windows unit test failure was resolved in #2474?

**Comment by gramidt** [comment]
@bogdandrutu - Was this meant to be closed? Were the permissions issues mentioned in comments resolved?

**Pull request closed without merging**